From cd2a1f10c5fb47b9e05918ec8cd71fc952ee5298 Mon Sep 17 00:00:00 2001
From: Andrzej Ostruszka <amo@semihalf.com>
Date: Thu, 4 Nov 2021 09:44:01 +0000
Subject: [PATCH 1/3] Include frame header in buffer length

Since we are capturing packets at L2 we should not be using ETH_DATA_LEN
but ETH_FRAME_LEN to include also dst/src MAC and EtherType.
---
 src/if-linux.c | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/src/if-linux.c b/src/if-linux.c
index e3cc257e..d0f982e1 100644
--- a/src/if-linux.c
+++ b/src/if-linux.c
@@ -1354,13 +1354,13 @@ bpf_open(struct interface *ifp, int (*filter)(struct interface *, int))
 	state = ipv4_getstate(ifp);
 	if (state == NULL)
 		goto eexit;
-	if (state->buffer_size < ETH_DATA_LEN) {
+	if (state->buffer_size < ETH_FRAME_LEN) {
 		void *nb;
 
-		if ((nb = realloc(state->buffer, ETH_DATA_LEN)) == NULL)
+		if ((nb = realloc(state->buffer, ETH_FRAME_LEN)) == NULL)
 			goto eexit;
 		state->buffer = nb;
-		state->buffer_size = ETH_DATA_LEN;
+		state->buffer_size = ETH_FRAME_LEN;
 		state->buffer_len = state->buffer_pos = 0;
 	}
 
-- 
2.34.0.rc0.344.g81b53c2807-goog

